src/tests/functional/inference_engine/ir_serialization/deterministicity.cpp

```diff
@@ -92,3 +94,44 @@ TEST_F(SerializationDeterministicityTest, SerializeToBlob) {
     ASSERT_TRUE(expected.layerCount() == result.layerCount());
     ASSERT_TRUE(expected.getInputShapes() == result.getInputShapes());
 }
+
+class DeterministicityTest : public ::testing::Test {
+protected:
+    std::string test_name = ::testing::UnitTest::GetInstance()->current_test_info()->name();
+    std::string m_out_xml_path = test_name + ".xml";
+    std::string m_out_bin_path = test_name + ".bin";
+
+    void TearDown() override {
+        std::remove(m_out_xml_path.c_str());
+        std::remove(m_out_bin_path.c_str());
+    }
+
+    void Run(const std::string& model_path) {
+        ov::runtime::Core core;
+        auto base_function = core.read_model(model_path);
+        EXPECT_NE(base_function, nullptr);
+
+        auto serialize = ov::pass::Serialize(m_out_xml_path, m_out_bin_path);
+        serialize.run_on_function(base_function);
+        auto serialized_func = core.read_model(m_out_xml_path);
+
+        auto base_it = base_function->get_parameters().begin();
+        auto serialized_it = serialized_func->get_parameters().begin();
+        while (base_it != base_function->get_parameters().end() || serialized_it != serialized_func->get_parameters().end()) {
+            ASSERT_EQ(base_function->get_parameter_index(*base_it), serialized_func->get_parameter_index(*serialized_it));
+            ASSERT_STREQ((*base_it)->get_friendly_name().c_str(), (*serialized_it)->get_friendly_name().c_str());
+            ++base_it;
+            ++serialized_it;
+        }
+    }
+};
+
+TEST_F(DeterministicityTest, IRInputsOrderSameAsInONNX) {
+    const std::string model_path = CommonTestUtils::getModelFromTestModelZoo(IR_SERIALIZATION_MODELS_PATH "few_inputs.onnx");
```

Comment:
Maybe we can create Model in place without external files

Reply:
Replaced with ov::Model generated in runtime